tests/testthat/test-yearmonth.R

```diff
@@ -12,6 +12,17 @@ test_that("input types for yearmonth()", {
   expect_identical(yearmonth(c(596, 576)), expected)
 })
 
+test_that("%Ym%M format for yearmonth()", {
+  expect_identical(yearmonth("2018 M01"), yearmonth("2018 Jan"))
+  expect_identical(yearmonth("2018 M1"), yearmonth("2018 Jan"))
+  expect_identical(yearmonth("2018mon01"), yearmonth("2018 Jan"))
+  expect_identical(yearmonth("2018 month 12"), yearmonth("2018 Dec"))
+  expect_error(yearmonth("201M1"))
+  expect_error(yearmonth("2018M13"))
+  expect_error(yearmonth("201M811"))
+  expect_error(yearmonth(c("2018M1", "2018 Feb", "2018M3")))
```

Comment:
Actually I think they are valid inputs, although a mixture of styles. I'll leave comments above to allow for this.